src/pushsource/_impl/backend/koji_source.py

```diff
@@ -504,6 +514,110 @@ def _push_items_from_container_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_vmi_build(self, nvr, meta):
+        LOG.debug("Looking for virtual machine images on %s, %s", nvr, meta)
+
+        if not meta:
+            message = "Virtual machine image build not found in koji: %s" % nvr
+            LOG.debug(message)
+            raise ValueError(message)
+
+        # VMI has the {'typeinfo': {'image':{}}} on extra
+        extra = meta.get("extra") or {}
+        typeinfo = extra.get("typeinfo") or {}
+        image = typeinfo.get("image")
+
+        # Container images have the key `container_koji_task_id` set on extras:
+        # https://github.com/containerbuildsystem/atomic-reactor/blob/master/docs/koji.md?plain=1#L79
```

Comment:
Minor: I think you should be linking to a specific commit or tag.  You have linked to the file in 'master' branch which can be changed at any time.  As soon as anyone inserts more lines prior to line 79, or renames or deletes the file, the link will be broken (and there will not even be an entirely reliable way to find out what this link was supposed to point to since there is no way to determine what 'master' in this repo pointed to at the time this commit was created).

Reply:
> and there will not even be an entirely reliable way to find out what this link was supposed to point to

It is true that commit are better but there is a reliable way how to get get to the correct line.
One would need to check date when the comment was added here(elease-engineering/pushsource) and use git "archeology" to checkout state of referenced repository at that date. It is not super ideal UX and also rely on disabled force push to master/main branch (which is de-facto standard in most of upstream projects)

On the other hand replacing  branch with commit is trivial as well `s!blob/master!blob/87155e0829b244192201b16ddbb693da6cea7811!`

Reply:
> there is a reliable way how to get get to the correct line. One would need to check date when the comment was added here(elease-engineering/pushsource) and use git "archeology" to checkout state of referenced repository at that date. 

How would you do it?  I can obtain the commit or author date of this commit in pushsource repo.  Wouldn't I then need github to tell me, "in containerbuildsystem/atomic-reactor, at time X, branch `master` pointed at commit Y" ?

I am aware that:

- commits have a commit date and author date, but this has no direct relationship to when a remote branch was pointing at a particular commit.
- locally, git reflog knows the history of the refs in my repo.  It can tell me what `master` pointed to at time X.  But this is only my local refs, which again has no direct relationship to the remote branches.

If you do know a reliable way to figure out what a remote repo's branch was pointing to at a specific time, please share as I'm interested to learn it.